src/mono/mono/mini/simd-intrinsics.c

```diff
@@ -2695,24 +2713,43 @@ emit_sri_vector (MonoCompile *cfg, MonoMethod *cmethod, MonoMethodSignature *fsi
 		}
 		return ins;
 	}
+	case SN_IsNegativeInfinity:
 	case SN_IsPositiveInfinity: {
 		if (!is_element_type_primitive (fsig->params [0]))
 			return NULL;
 		if (arg0_type == MONO_TYPE_R4) {
 			guint32 value[4];
 
-			value [0] = 0x7F800000;
-			value [1] = 0x7F800000;
-			value [2] = 0x7F800000;
-			value [3] = 0x7F800000;
+			if (id == SN_IsNegative)
```

Comment:
will id ever be SN_IsNegative here?

Reply:
No, this should've been `SN_IsNegativeInfinity` and I think there was just a copy paste/error. Will fix